package.json

```diff
@@ -10,51 +10,27 @@
 		"markdownv2"
 	],
 	"license": "MIT",
-	"repository": "EdJoPaTo/telegram-format",
+	"repository": {
+		"type": "git",
+		"url": "https://github.com/EdJoPaTo/telegram-format.git"
+	},
 	"author": {
 		"name": "EdJoPaTo",
-		"email": "telegram-format-npm-package@edjopato.de",
+		"email": "telegram-format-typescript-package@edjopato.de",
 		"url": "https://edjopato.de"
 	},
 	"scripts": {
-		"build": "del-cli dist && tsc",
-		"prepack": "npm run build",
-		"test": "tsc --sourceMap && xo && c8 --all ava"
+		"prepare": "deno2node",
+		"test": "echo use deno test && exit 1"
```

Comment:
```suggestion
		"test": "deno test"
```

Reply:
Hm… this gets me thinking. On one side I don’t want to have much npm stuff when it’s a Deno repo. Also it just fails without Deno being available. On the other side it’s a common thing to just use npm test and it works. Also npm test outputs the command run so it would be obvious that Deno test is being run.

Reply:
I think I will stay with the echo to be clear about the tests being on deno. Porting the tests to Node.js and them run them would create a useful `npm test` but otherwise its just misleading in my opinion.

Thank you for bringing this up! I haven't thought that much about it before.

Reply:
Running tests on Node.js requires transpiling them first, so `"postprepare"` is a better fit.

Reply:
https://docs.npmjs.com/cli/v10/commands/npm-install-test is a convenient way to run `"prepare"`, `"postprepare"` and `"test"`.

Reply:
For me its not only about the scripts involved here. It also requires manually writing tests for Node.js using node:test for example. As far as I can see its done manually within the deno2node repository which is more effort for the same thing. Like why should I test that `Array.prototype.filter` also works on Node.js or a browser. I want to assume that that's the case.

I want to use this for code which isn't dependent on the environment (Deno, Node.js, Browser). Then `deno test` should be enough in my opinion.

When the code is dependent on the platform (like filesystem or something like that) its probably a better idea to move that into its own library especially written for that specific environment. (Or even better: write the library in a way that the user can decide on that which removes the dependency from the library in the first place.)

The only case that I can see are dependencies which might have wrong import code on the Node.js part (`deps.node.ts`). Otherwise the code should be identical in my opinion.

Maybe I am missing something. Maybe my personal assumptions on where to use this are different. Currently it seems like more manual overhead compared to its benefits for me. My goal is a clean platform independent JavaScript / TypeScript library available on Deno & Node.js without much overhead.

I like that we are talking about it to figure out these specifics!

Reply:
Deno implements `node:test` and `node:assert`. But my point is, there's no reason for `npm test` and `npm install-test` to not work.

Reply:
Yeah, but I still need to manually implement tests specific for Node.js? `Deno.test` on Deno and `node:assert` on Node.js only to test that the same V8 engine runs the same code?

Running `npm test` then would hide the fact that it’s not tested on Node.js explicitly? It’s still tested within the Deno environment which is the main environment here which people should use when working with this repository?

Reply:
> I still need to manually implement tests specific for Node.js?

No. `node:test` and `node:assert` work on both runtimes. `deno2node` takes care of other differences.

Reply:
Guess know we know where I misunderstood you 🙃
It still feels wrong to use something with `node:` on Deno but it’s a low effort solution for tests on both platforms.